lfs.c

```diff
@@ -3436,11 +3530,15 @@ static int lfs_commitattr(lfs_t *lfs, const char *path,
     return lfs_dir_commit(lfs, &cwd, LFS_MKATTRS(
             {LFS_MKTAG(LFS_TYPE_USERATTR + type, id, size), buffer}));
 }
+#endif
 
 int lfs_setattr(lfs_t *lfs, const char *path,
         uint8_t type, const void *buffer, lfs_size_t size) {
     LFS_TRACE("lfs_setattr(%p, \"%s\", %"PRIu8", %p, %"PRIu32")",
             (void*)lfs, path, type, buffer, size);
+#ifdef LFS_READONLY
+    return LFS_ERR_NOSYS;
```

Comment:
It looks like you are using LFS_ERR_NOSYS for any functions exposed in lfs.h. I think it would be better to also conditionally omit the definitions in the header file.

Build errors are usually more helpful than runtime errors, and worst case the user can provide an additional stub function that returns LFS_ERR_NOSYS.

Reply:
Right! I did that because it's the way SPIFFS implemented their read-only support. Not that we have to copy that behavior, of course, but that's where I'm migrating from.
I will rework this to simply omit the definitions as well.
I'll take into account the other remarks as well, and update the PR.